proto/backend.proto

```diff
@@ -110,14 +110,21 @@ message QueryDataRequest {
 }
 
 message QueryDataResponse {
-  // Arrow encoded DataFrames
-  // Each frame encodes its own: Errors, meta, and refId
-  repeated bytes frames = 1;
+  // Map of refId to response
+  map<string,DataResponse> responses = 1;
 
   // Additional response metadata
   map<string,string> metadata = 2;
 }
 
+message DataResponse {
+  // Arrow encoded DataFrames
+  // Frame has its own meta, warnings, and repeats refId
+  repeated bytes frames = 1;
+  string error = 2;
+  bytes queryMeta = 3;
```

Comment:
did we agree to put JSON in the name when this represents JSON bytes?

Reply:
renamed to jsonMeta